src/sidebar/util/url.js

```diff
@@ -8,10 +8,11 @@
  *     {url: '/things/foo', params: {q: 'bar'}}
  *
  * @param {string} url
- * @param {Record<string, any>} params
- * @return {{ url: string, params: Record<string, any>}}
+ * @param {Record<string, string>} params
+ * @return {{ url: string, params: Record<string, string>}}
```

Comment:
The `params` name is prone to confusion. I would suggest a reimplementation of this function:

```suggestion
 * @return {{ url: string, unusedParams: Record<string, string>}}
 */
export function replaceURLParams(url, params) {
  /** @type {Record<string, string>}} */
  const unusedParams = {};
  Object.entries(params).forEach(([key, value]) => {
    const urlParam = `:${key}`;
    if (url.includes(urlParam)) {
      url = url.replace(urlParam, encodeURIComponent(value));
    } else {
      unusedParams[key] = value;
    }
  });

  return { url, unusedParams };
}
```

For this or another PR.

Reply:
I agree. I'll make this change in a separate commit.